lib/fireworq.go

```diff
@@ -85,6 +142,23 @@ func (p FireworqPlugin) FetchMetrics() (map[string]float64, error) {
 	m["active_nodes"] = float64(sum.ActiveNodes)
 	m["active_nodes_percentage"] = float64(sum.ActiveNodes*100) / float64(len(stats))
 
+	for q, s := range stats {
+		if s.ActiveNodes >= 1 {
+			// Escape queue name to confirm to metric name specification.
+			// See also: https://mackerel.io/ja/api-docs/entry/host-metrics#post-graphdef
+			r := regexp.MustCompile("[^-a-zA-Z0-9_]")
+			q = r.ReplaceAllString(q, "-")
+
+			if job, err := p.fetchMostDelayedJob(q); err == nil {
+				var delay float64
+				if job != nil {
+					delay = float64(time.Since(job.NextTry).Seconds())
+				}
+				m[fmt.Sprintf("queue.delay.%s", q)] = delay
```

Comment:
[memo]

The delay will be 0 if there is no job because it means the queue is empty.